src/Microsoft.Framework.ConfigurationModel.Json/JsonConfigurationExtension.cs

```diff
@@ -1,12 +1,37 @@
 // Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.IO;
+
 namespace Microsoft.Framework.ConfigurationModel
 {
     public static class JsonConfigurationExtension
     {
         public static IConfigurationSourceRoot AddJsonFile(this IConfigurationSourceRoot configuration, string path)
         {
+            if (!Path.IsPathRooted(path))
+            {
+                if (configuration.BasePath == null)
+
+                {
+                    throw new InvalidOperationException(
+                        string.Format("The {0} is not rooted and the {1} is null in this {2}",
```

Comment:
1. this doesn't tell the developer much.  suggest `Unable to resolve path '{0}'; construct this {2} with a non-null {1}.` where `{0}` is the `path` value and not `nameof(path)`
2. this project has resources; new messages should be there, not inline